lib/rpmdb.c

```diff
@@ -731,13 +731,14 @@ int rpmdbCountPackages(rpmdb db, const char * name)
 }
 
 /**
- * Attempt partial matches on name[-version[-release]][.arch] strings.
+ * Attempt partial matches on name[-version[-release]][/disttag][.arch] strings.
```

Comment:
Why is there a `/` here? I don't think we'd reasonably consider a `/` for a separator...

Reply:
I would vote for `N@D:E-V-R.A` probably

Reply:
Do you really think DistTag as a super-epoch makes sense? Historically, usage of DistTag was incorporated at the end of the comparison: `N-E:V-R-D.A`.

Reply:
I decided to use `/` because it does not make collision with other part of NEVRA, and it is easy to parse (see the code).

@ignatenkobrain  you really mean `N@D:E-V-R.A`? It breaks backward compatibility. More sane is `N[@D]-[[E:][V[-R]]][.A]`. But I would prefer to stay with my proposal for a while.

@Conan-Kudo do you have example where `N-E:V-R-D.A` is using?

Reply:
I don't know should we transfer this discussion to #594, this is still discussible.

Please, tell what you think about `rpmlib(HandleDistTag)`.

Reply:
@wladmis All packages in OpenMandriva Lx3 work this way.

As for `rpmlib(HandleDistTag)`, I'd actually rather call it `rpmlib(DistTagInVersionComparison)`.

Reply:
The version string should be `4.15-1`

Reply:
@wladmis Actually, OMV Lx3 packages use `N-E:V-R:D.A`...

Reply:
@Conan-Kudo so you think of new feature (lets name it `rpmlib(DistTagInVersionComparison)`, it's not principal) is a good idea. @ignatenkobrain your opinion?

So OMV Lx3 uses `N-E:V-R:D.A` but it may do collision cause `:` is valid symbol for `release`. I will take a look at OMV Lx3. I still prefer my variant, but I can change my mind.